tools/c7n_gcp/c7n_gcp/resources/appengine.py

```diff
@@ -0,0 +1,146 @@
+        parent_spec = {
+            'resource': 'appengine-app',
+            'child_enum_params': {
+                ('id', 'appsId')
+            }
+        }
+
+        @staticmethod
+        def get(client, resource_info):
+            name_param_re = re.compile('apps/(.*?)/authorizedCertificates/(.*)')
+            apps_id, cert_id = name_param_re.match(resource_info['resourceName']).groups()
+            return client.execute_query('get', {'appsId': apps_id,
+                                                'authorizedCertificatesId': cert_id})
+
+
+@resources.register('appengine-domain')
+class AppEngineDomain(ChildResourceManager):
+
+    class resource_type(ChildTypeInfo):
+        service = 'appengine'
+        version = 'v1'
+        component = 'apps.authorizedDomains'
+        enum_spec = ('list', 'domains[]', None)
+        scope = None
+        id = 'id'
+        parent_spec = {
+            'resource': 'appengine-app',
+            'child_enum_params': {
+                ('id', 'appsId')
+            }
+        }
+
+
+@resources.register('appengine-domain-mapping')
+class AppEngineDomainMapping(ChildResourceManager):
+
+    def _get_parent_resource_info(self, child_instance):
+        return {'resourceName': re.compile(
+            '(apps/.*?)/domainMappings/.*').match(child_instance['name']).group(1)}
+
+    class resource_type(ChildTypeInfo):
+        service = 'appengine'
+        version = 'v1'
+        component = 'apps.domainMappings'
+        enum_spec = ('list', 'domainMappings[]', None)
+        scope = None
+        id = 'id'
+        parent_spec = {
+            'resource': 'appengine-app',
+            'child_enum_params': {
+                ('id', 'appsId')
+            }
+        }
+
+        @staticmethod
+        def get(client, resource_info):
+            name_param_re = re.compile('apps/(.*?)/domainMappings/(.*)')
+            apps_id, mapping_id = name_param_re.match(resource_info['resourceName']).groups()
+            return client.execute_query('get', {'appsId': apps_id,
+                                                'domainMappingsId': mapping_id})
+
+
+@resources.register('appengine-firewall-ingress-rule')
+class AppEngineFirewallIngressRule(ChildResourceManager):
+
+    def _get_parent_resource_info(self, child_instance):
+        return {'resourceName': 'apps/%s' %
+                                local_session(self.session_factory).get_default_project()}
+
+    class resource_type(ChildTypeInfo):
+        service = 'appengine'
+        version = 'v1'
+        component = 'apps.firewall.ingressRules'
+        enum_spec = ('list', 'ingressRules[]', None)
+        scope = None
+        id = 'id'
+        parent_spec = {
+            'resource': 'appengine-app',
+            'child_enum_params': {
+                ('id', 'appsId')
+            }
+        }
+
+        @staticmethod
+        def get(client, resource_info):
+            rule_re = re.compile('apps/(.*?)/firewall/ingressRules/(.*)')
```

Comment:
there is no point to compiling a regex if your throwing away the compilation. either store the compiled result as a class var, or ditch the compilation.